cmd/oapi-gen/gen.go

```diff
@@ -245,13 +248,33 @@ func fieldName(field *ast.Field, tag string) (string, bool) {
 }
 
 func docToString(cg *ast.CommentGroup) string {
-	s := cg.Text()
-	if s == "" {
-		return ""
+	lines := make([]string, 0, len(cg.List))
+	for _, c := range cg.List {
+		text := strings.TrimPrefix(c.Text, "//")
+		text = strings.TrimPrefix(text, "/*")
+		text = strings.TrimSuffix(text, "*/")
+		text = strings.TrimSpace(text)
+
+		// Skip directive lines
+		if strings.HasPrefix(text, "openapi:") {
+			continue
+		}
+
+		// Skip empty lines.
+		if text == "" {
+			continue
+		}
+
+		lines = append(lines, text)
 	}
+	return html.EscapeString(strings.Join(lines, "\n"))
```

Comment:
You are making your life really hard here:
```suggestion
	s := cg.Text()
	if s == "" {
		return ""
	}

	s = strings.TrimSuffix(s, "\n")
	s = strings.Join(strings.Split(s, "\n"), "\\n") // Escape newlines.
	return strings.TrimSpace(s)
```